tests/TestCase/ORM/RulesCheckerIntegrationTest.php

```diff
@@ -1047,4 +1047,197 @@ public function testCountOfAssociatedItems()
+        $rulesChecker->addDelete(
+            $rulesChecker->isNotLinkedTo('Comments', 'comments')
+        );
+
+        $article = $Articles->get(3);
+        $this->assertTrue($Articles->delete($article));
+    }
+
+    public function testIsLinkedToPassAssociationInstance()
+    {
+        $Comments = TableRegistry::get('Comments');
+        $Comments->belongsTo('Articles');
+        $Comments->association('Articles')->hasMany('Comments');
+
+        /* @var $rulesChecker \Cake\ORM\RulesChecker */
+        $rulesChecker = $Comments->rulesChecker();
+        $rulesChecker->addUpdate(
+            $rulesChecker->isLinkedTo($Comments->association('Articles'), 'articles')
+        );
+
+        $comment = $Comments->get(1);
+        $comment->dirty('comment', true);
+        $this->assertNotFalse($Comments->save($comment));
+    }
+
+    public function testIsNotLinkedToPassAssociationInstance()
+    {
+        $Articles = TableRegistry::get('Articles');
+        $Articles->hasMany('Comments');
+        $Articles->association('Comments')->belongsTo('Articles');
+
+        /* @var $rulesChecker \Cake\ORM\RulesChecker */
+        $rulesChecker = $Articles->rulesChecker();
+        $rulesChecker->addDelete(
+            $rulesChecker->isNotLinkedTo($Articles->association('Comments'), 'comments')
+        );
+
+        $article = $Articles->get(3);
+        $this->assertTrue($Articles->delete($article));
+    }
+
+    public function testIsLinkedToPassTableInstance()
+    {
+        $Comments = TableRegistry::get('Comments');
+        $Comments->belongsTo('Articles');
+        $Comments->association('Articles')->hasMany('Comments');
+
+        /* @var $rulesChecker \Cake\ORM\RulesChecker */
+        $rulesChecker = $Comments->rulesChecker();
+        $rulesChecker->addUpdate(
+            $rulesChecker->isLinkedTo($Comments->association('Articles')->target(), 'articles')
+        );
+
+        $comment = $Comments->get(1);
+        $comment->dirty('comment', true);
+        $this->assertNotFalse($Comments->save($comment));
+    }
+
+    public function testIsNotLinkedToPassTableInstance()
+    {
+        $Articles = TableRegistry::get('Articles');
+        $Articles->hasMany('Comments');
+        $Articles->association('Comments')->belongsTo('Articles');
+
+        /* @var $rulesChecker \Cake\ORM\RulesChecker */
+        $rulesChecker = $Articles->rulesChecker();
+        $rulesChecker->addDelete(
+            $rulesChecker->isNotLinkedTo($Articles->association('Comments')->target(), 'comments')
+        );
+
+        $article = $Articles->get(3);
+        $this->assertTrue($Articles->delete($article));
+    }
+
+    public function testIsLinkedToOmittingFieldArgumentCausesNoErrorToBeSet()
+    {
+        $Comments = TableRegistry::get('Comments');
+        $Comments->belongsTo('Articles');
+        $Comments->association('Articles')->hasMany('Comments');
+
+        $Comments->save($Comments->newEntity([
```

Comment:
@ionas  Not sure what exactly your are referring to? Generally checking rules and creating/updating/deleting is happening in a transaction by default, yes.